crates/iota-sdk-ffi/src/lib.rs

```diff
@@ -137,3 +137,43 @@ pub fn hex_encode(input: &[u8]) -> String {
 pub fn hex_decode(input: String) -> crate::error::Result<Vec<u8>> {
     Ok(hex::decode(input)?)
 }
+
+#[macro_export]
+macro_rules! export_record_enum_bcs_conversion {
```

Comment:
This makes me sad. Users of the lib will obviously not understand why some types have methods and some don't, and this will be confusing. But I don't know what to do other than just make everything an object, and that sucks. 😤

Reply:
okay, so changed everything to free fns, for objects and records/enums, so for every type it's now the same consistent API.